z2/src/chain/node.rs

```diff
@@ -548,6 +548,54 @@ impl ChainNode {
         ctx.insert("api_servers", &toml_servers.to_string());
         ctx.insert("enable_ots_indices", &enable_ots_indices);
 
+        if let Some(checkpoint_url) = self.chain.checkpoint_url() {
+            if self.role == NodeRole::Validator {
+                let checkpoint_file = checkpoint_url.rsplit('/').next().unwrap_or("");
+                ctx.insert("checkpoint_file", &format!("/{}", checkpoint_file));
+
+                let checkpoint_hex_block =
+                    crate::utils::string_decimal_to_hex(&checkpoint_file.replace(".dat", ""))?;
+
+                let json_response = self
+                    .chain
+                    .run_rpc_call(
+                        "eth_getBlockByNumber",
+                        &Some(format!("[\"{}\", false]", checkpoint_hex_block)),
+                        30,
+                    )
+                    .await?;
```

Comment:
@DrZoltanFazekas Do you think its okay for the infra tooling to obtain the checkpoint's block hash itself like this or should we force the user to specify it themselves from a source they trust?